proto/options.proto

```diff
@@ -41,6 +41,7 @@ enum ShapeFormat {
   polyline6 = 0;
   polyline5 = 1;
   geojson = 2;
+  no_shape = 3;  // to omit returning the geometry, e.g. default in matrix
```

Comment:
nit: 
```suggestion
  none = 3;  // to omit returning the geometry, e.g. default in matrix
```

in worker.cc dont we have to protect against this value when its not a matrix action

Reply:
let's see if that works. there's already a `none` enum member and the way protobuf works is they are siblings, pretty sure that'll clash. but let's see:)

Reply:
> in worker.cc dont we have to protect against this value when its not a matrix action

I did that for `/height` (you probably didn't get to worker.cc yet..), which is the only documented use of `shape_format`. We do use it for OSRM as well, but it's not documented and other than mapbox I'm sure no one knows. I even added a bit of code to use `no_shape` for OSRM as well. Eventually we should enable `shape_format` also for the default Valhalla format. I find it kinda useful to disable getting the geometry from a route call, same as the narrative, it's the second largest payload item and some might not be interested in the geometry.

Reply:
yeah that's what I meant @kevinkreiser :
```
[build] options.proto:44:3: Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it
```

stupid protobuf..

should we just leave `no_shape`, it's short enough and very exclicit.

Reply:
yeah i thought this might be the case but it was worth asking.

shape_format=no_shape i just find redundant, but its fine.

Reply:
though i do think we can change the nesting to accomplish this if we really want

Reply:
> shape_format=no_shape i just find redundant, but its fine.

it's necessary for omitting `shape` from a route response no? at least I don't think we should change the default for osrm format. or how else would you do it?

Reply:
oh jeez im sorry i just meant the naming is redundant i get that we need a way to say we dont want shape. `shape_format=shapity_shape_shape_shaperino` zum beispiel :wink: we could name it `disabled` or `off` or something but i dont care that much im just very into language stuff